src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-utils.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    EDITED_FIELD,
+    FILTERS,
+    ID,
+    NAME,
+    PROPERTY_NAME_FIELD,
+    VALUE_FIELD,
+} from '../../../../../utils/field-constants';
+import yup from 'components/utils/yup-config';
+import { Schema } from 'yup';
+import { Assignment, DataType } from './assignment.type';
+import { FIELD_OPTIONS } from './assignment-constants';
+
+export const getDataType = (fieldName?: string | null) => {
+    return Object.values(FIELD_OPTIONS).find((fieldOption) => fieldOption.id === fieldName)?.dataType;
+};
+
+// ("undefined" is accepted here in RHF, but it conflicts with MUI behaviour which does not like undefined values)
+export const getAssignmentInitialValue = () => ({
+    [FILTERS]: [],
+    [EDITED_FIELD]: null,
+    [PROPERTY_NAME_FIELD]: null,
+    [VALUE_FIELD]: null,
+});
+
+export function getAssignmentsSchema() {
+    return yup
+        .array()
+        .of(
+            yup.object().shape({
+                [FILTERS]: yup
+                    .array()
+                    .of(
+                        yup.object().shape({
+                            [ID]: yup.string().required(),
+                            [NAME]: yup.string().required(),
+                        })
+                    )
+                    .required()
+                    .min(1, 'FieldIsRequired'),
```

Comment:
no translation for  FieldIsRequired ?

Reply:
YupRequired is a newer key, I think should switch to the newer

Reply:
DONE